backend-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/TorWalletClient.kt

```diff
@@ -0,0 +1,129 @@
+package cash.z.ecc.android.sdk.internal.model
+
+import cash.z.wallet.sdk.internal.rpc.Service
+import co.electriccoin.lightwallet.client.PartialTorWalletClient
+import co.electriccoin.lightwallet.client.model.BlockHeightUnsafe
+import co.electriccoin.lightwallet.client.model.BlockIDUnsafe
+import co.electriccoin.lightwallet.client.model.LightWalletEndpointInfoUnsafe
+import co.electriccoin.lightwallet.client.model.RawTransactionUnsafe
+import co.electriccoin.lightwallet.client.model.Response
+import co.electriccoin.lightwallet.client.model.SendResponseUnsafe
+import co.electriccoin.lightwallet.client.model.TreeStateUnsafe
+import com.google.protobuf.kotlin.toByteString
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.withContext
+
+class TorWalletClient private constructor(
+    private var nativeHandle: Long?,
+) : PartialTorWalletClient {
+    private val semaphore = Mutex()
+
+    override suspend fun dispose() =
+        withContext(Dispatchers.IO) {
+            semaphore.withLock {
```

Comment:
Can `withContext(Dispatchers.IO)` block? If so, I think there's a deadlock here compared to `TorLwdConn`:
- `dispose()` acquires `Dispatchers.IO` first, then `semaphore`.
- `execute` acquires `semaphore` first, then `Dispatchers.IO`.

Instead I think everywhere inside `TorWalletConn` should do things the same way round.